microsoft/seismic-deeplearning
# Azureml train pipeline

[Still requires documentation about running with dutchf3 data specifically.]

This PR adds configurable Azure ML training pipelines to Seismic DeepLearning. Each pipeline is configured via the pipeline json and environment variables. See the README added in this PR (azureml_pipelines folder) for more information. 

- A few arguments had to be added to train.py: input and output. These allow the pipeline to pass in mounted Blob storage locations to be used as a regular file system. 

- azureml_requirements.txt file was added to the training directory. This file holds all dependencies for train.py so they can be installed on the compute in Azure ML. 

- An empty __init__.py was added to the segmentation/dutchf3 module. I was unable to install cv_lib and access cv_lib.segmentation.dutchf3.utils without this when installing from github (note: to get this version to work, I am referencing this feature branch in the requirements. It should be changed to microsoft:staging before merging)

- There are two scripts added: kickoff_train_pipeline.py & cancel_run.py. Kickoff_train shows how to run an Azure ML train pipeline & cancel_run is a development script for cancelling the run instead of letting it take resources to completion.

- The train_pipeline inherits from base_pipeline which would be a helpful abstraction for future addition of an inference pipeline

- Integration tests for kicking off a train pipeline are added. These are not currently run in a build pipeline because they require communication with an Azure ML Service instance

- Requirements for kicking off the pipeline are added to interpretation's requirements.txt


To Test:

1. Prepare dutchf3 data and upload to blob storage
2. Create an instance of Azure ML service
3. Follow instructions for setting up the environment variables and kicking off a pipeline in the pipelines README
4. Verify that training runs successfully (currently taking ~2 hours for 1 epoch)
